README.md

```diff
@@ -0,0 +1,303 @@
+        $object = $event->getObject();        // Get the affected DataObject
+        $class = $event->getObjectClass();    // Get the class name
+        $operation = $event->getOperation();  // Get the operation type
+        $version = $event->getVersion();      // Get version number (if versioned)
+        $member = $event->getMember();        // Get the Member who performed the action
+        $time = $event->getTimestamp();       // Get when the event occurred
+    },
+    [DataObject::class]
+)->selfRegister();
+```
+
+`DataObjectEvent` is configured to be serializable so it can easily be stored for later use.
+
+Note that `DataObjectEvent` doesn't store the actual DataObject instance that caused the event to be fired. 
+`DataObjectEvent::getObject()` will refetch the latest version of the DataObject from the database ... which will 
+return `null` if the DataObject has been deleted.
+
+`DataObjectEvent::getObject(true) will attempt to retrieve the exact version of the DataObject that fired the event,
+assuming it was versioned.
+
+## Testing Your Events
+
+### Writing Event Tests
+
+When testing your event listeners, you'll need to:
+1. Dispatch your events
+2. Run the event loop
+3. Assert the expected outcomes
+
+Here's an example test:
+
+```php
+use Revolt\EventLoop;
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\Core\Injector\Injector;
+use ArchiPro\Silverstripe\EventDispatcher\Service\EventService;
+
+class MyEventTest extends SapphireTest
+{
+    public function testMyCustomEvent(): void
+    {
+        // Create your test event
+        $event = new MyCustomEvent('test message');
+        
+        // Get the event service
+        $service = Injector::inst()->get(EventService::class);
+        
+        // Add your test listener ... or if you have already
+        $wasCalled = false;
+        $service->addListener(
+            MyCustomEvent::class, 
+            [MyCustomEventListener::class, 'handleEvent']
+        );
+        
+        // Dispatch the event
+        $service->dispatch($event);
+        
+        // Run the event loop to process events
+        EventLoop::run();
+        
+        // Assert your listener was called
+        $this->assertTrue(
+            MyCustomEventListener::wasCalled(),
+            'Assert some side effect of the event being handled'
+        );
+    }
+}
+```
+
+### Disabling event dispatching
+
+You can disable event dispatching for test to avoid side affects from irrelevant events that might be fired while 
+scaffolding fixtures.
+
+Call `EventService::singleton()->disableDispatch()` to disable event dispatching while setting up your test.
+
+When you are ready to start running your test, call `EventService::singleton()->enableDispatch()` to start listening for
+events again.
+
+### Important Testing Notes
+
+- Always remember to run `EventLoop::run()` after dispatching events
+- Events are processed asynchronously, so assertions should happen after running the event loop
```

Comment:
This may change should you want to explicitly deal with futures. You can await them without manually executing the loop.

```
$result = $service->dispatch($event)->await();
```

Reply:
P.S await() **will** block depending on where and how you use it.